src/ui/public/courier/data_source/_abstract.js

```diff
@@ -382,6 +384,11 @@ export default function SourceAbstractFactory(Private, Promise, PromiseEmitter)
             recurse(agg.aggs || agg.aggregations);
           });
         }(flatState.body.aggs || flatState.body.aggregations));
+
+        if (flatState.body._source && flatState.body.fielddata_fields) {
+          const filter = fieldWildcardFilter(flatState.body._source.exclude);
+          flatState.body.fielddata_fields = flatState.body.fielddata_fields.filter(filter);
+        }
```

Comment:
I'm pretty sure this block could be moved up to where `fielddata_fields` is initialized: https://github.com/elastic/kibana/blob/90894ea7aeda5a74a11c7b990e0f56ef23c99a83/src/ui/public/courier/data_source/_abstract.js#L295-L295